src/Server.js

```diff
@@ -23,69 +24,72 @@ function getResultsFromCache() {
   return results;
 }
 
+const lintFile = (file) => {
+  process.send({file: file});
+  if (eslint.isPathIgnored(path.join(ROOT_DIR, file))) {
+    return;
+  }
+  filesToProcess++;
+  runLint({ cwd: ROOT_DIR }, [file])
+    .then(function(results) {
+      const result = results[0];
+      if (result) {
+        delete result.source;
+        cache[result.filePath] = result;
+      }
+      filesToProcess--;
+    })
+    .catch(e => console.error(e));
+}
+
+const lintAllFiles = (files) => {
+  files.map((file, _) => {
+    lintFile(file);
+  });
+}
+
 export default class Server {
-  constructor() {
-    this._setupWatcher(WatchmanClient);
+  constructor(port, numWorkers, pathsToLint) {
+    this.port = port;
+    this.numWorkers = numWorkers;
+    this.pathsToLint = pathsToLint;
+
+    this._setupSaneWatcher();
+
     const server = dnode({
       status: (param, cb) => {
         if (filesToProcess === 0) {
           return cb(getResultsFromCache());
         }
 
-        setInterval(
-          () => {
-            if (filesToProcess > 0) {
-              // console.log(`Linting... ${filesToProcess} left`);
-            } else {
-              cb(getResultsFromCache());
-            }
-          },
-          1000
-        );
+        process.send({filesToProcess: filesToProcess});
```

Comment:
so this is just for that initial run right? since after that the process is forked and the IPC channel is closed